ros/processor/inventory_events_consumer.py

```diff
@@ -119,42 +119,67 @@ def process_system_details(self, msg):
         """ Store new system information (stale, stale_warning timestamp) and return internal DB id"""
         host = msg['host']
         with app.app_context():
-            try:
-                account = get_or_create(
-                    db.session, RhAccount, 'account',
-                    account=host['account'],
-                    org_id=host.get('org_id')
-                )
-
+            if msg.get('type') == 'updated':
                 system_fields = {
-                    "tenant_id": account.id,
                     "inventory_id": host['id'],
                     "display_name": host['display_name'],
                     "fqdn": host['fqdn'],
                     "cloud_provider": host['system_profile']['cloud_provider'],
                     "stale_timestamp": host['stale_timestamp'],
                     "operating_system": host['system_profile']['operating_system'],
                 }
-                system = get_or_create(db.session, System, 'inventory_id', **system_fields)
-
-                # Commit changes
-                db.session.commit()
-                processor_requests_success.labels(
-                    reporter=self.reporter, org_id=account.org_id
-                ).inc()
-                LOG.info(
-                    f"{self.prefix} - Refreshed system {system.inventory_id} ({system.id}) "
-                    f"belonging to account: {account.account} ({account.id}) and org_id: {account.org_id}"
-                )
-            except Exception as err:
-                processor_requests_failures.labels(
-                    reporter=self.reporter, org_id=account.org_id
-                ).inc()
+                system = update_system_record(db.session, **system_fields)
```

Comment:
There is a scenario where system already been registered and enabled ros later. In such scenario inventory service will send out message with `updated` event type as system already exists on Inventory side.  With this gate condition, ROS will reject such systems.
This needs to be handled.